extensions/interactions/TextInput/TextInput.js

```diff
@@ -81,3 +82,56 @@ oppia.directive('oppiaShortResponseTextInput', [
     };
   }
 ]);
+
+oppia.factory('textInputRulesService', ['$filter', function($filter) {
+  return {
+    Equals: function(answer, inputs) {
+      var normalizedAnswer = $filter('normalizeWhitespace')(answer);
+      var normalizedInput = $filter('normalizeWhitespace')(inputs.x);
+      return normalizedAnswer.toLowerCase() == normalizedInput.toLowerCase();
+    },
+    FuzzyEquals: function(answer, inputs) {
+      var normalizedAnswer = $filter('normalizeWhitespace')(answer);
+      var answerString = normalizedAnswer.toLowerCase();
+
+      var normalizedInput = $filter('normalizeWhitespace')(inputs.x);
+      var inputString = normalizedInput.toLowerCase();
+
+      if(inputString == answerString) return true;
+      var d = [];
+      for(var i = 0; i <= inputString.length; i++){
+        d.push([i]);
+      }
+      for(var j = 0; j <= answerString.length; j++){
+        d[0].push(j);
+      }
+      for(i = 1; i <= inputString.length; i++){
```

Comment:
Just tested it and d[inputString.length][answerString.length] seems to be 1 as intended.

The implementation from normalised_string.py makes use of list comprehension functions in python, and the array indexing of -1. I tried translating the logic directly to javascript, then realised that while it was space efficient, it was too messy with too many if conditions and could be expressed more simply, so I decided to write my own instead.